docs/content/docs/security/running-cluster.md

```diff
@@ -0,0 +1,193 @@
+
+Incorporating security in this mode involves the following steps:
+1. Add [security-related configuration options]({{< ref "docs/deployment/config" >}}#auth-with-external-systems)) 
+   to the Flink configuration file (`flink-conf.yaml`) on all cluster nodes.
+2. Ensure that the Kerberos keytab file exists at the path indicated by the `security.kerberos.login.keytab`
+   setting on all cluster nodes.
+3. Deploy the Flink cluster as normal.
+
+{{< hint info >}}
+`HADOOP_TOKEN_FILE_LOCATION` is an important environment variable that specifies the location of your 
+credential files/tokens. The service instances uses this location to find the file to load the credentials 
+and run the tasks. Launched containers must load the delegation tokens from this location, and use them 
+(including renewals) until they can no longer be renewed.
+{{< /hint >}}
+
+You need to have an appropriate Java keystore and truststore accessible from each node in the Flink 
+cluster. For standalone setups, this means copying the files to each node, or adding them to a shared 
+mounted directory. For externally facing REST endpoints, the common name or subject alternative names 
+in the certificate should match the node's hostname and IP address.
+
+Keystores and truststores can be generated using the [keytool utility](https://docs.oracle.com/javase/8/docs/technotes/tools/unix/keytool.html).
+
+## Running a Secure Flink Cluster in Kubernetes
+
+Flink’s native Kubernetes integration allows you to directly deploy Flink on a running Kubernetes cluster. 
+You can learn more [here]({{< ref "docs/deployment/resource-providers/native_kubernetes" >}}).
+
+Incorporating security in this mode involves the following steps:
+1. Add [security-related configuration options]({{< ref "docs/deployment/config" >}}#auth-with-external-systems))  
+   to the Flink configuration file (`flink-conf.yaml`) on the client node.
+2. Ensure that the Kerberos keytab file exists at the path as indicated by the `security.kerberos.login.keytab` 
+   on the client node.
+3. Deploy the Flink cluster as normal.
+
+### Setting up Kerberos
+
+When deploying Flink on Kubernetes, the Kerberos keytab file is automatically copied from the client 
+node to the Flink containers.
+
+To enable Kerberos authentication, the Kerberos configuration file is also required. This file should 
+be present in the cluster nodes / classpath. If you want Flink to upload a new file, you need to configure 
+the `security.kerberos.krb5-conf.path` setting to indicate the path of the Kerberos configuration file
+and Flink will copy this file to its containers/pods.
+
+{{< hint info >}}
+`HADOOP_TOKEN_FILE_LOCATION` is an important environment variable that specifies the location of your
+credential files/tokens. The service instances uses this location to find the file to load the credentials
+and run the tasks. Launched containers must load the delegation tokens from this location, and use them
```

Comment:
`Launched containers can load the delegation tokens from this location` because Flink can fetch it in the initial phase too.